lms/djangoapps/course_api/blocks/transformers/block_completion.py

```diff
@@ -0,0 +1,82 @@
+"""
+Block Completion Transformer
+"""
+
+from lms.djangoapps.completion.models import BlockCompletion
+from openedx.core.djangoapps.content.block_structure.transformer import BlockStructureTransformer
+
+
+class BlockCompletionTransformer(BlockStructureTransformer):
+    """
+    Keep track of the completion of each block within the block structure.
+    """
+    WRITE_VERSION = 1
+    READ_VERSION = 1
+    BLOCK_COMPLETION = 'block_completion'
```

Comment:
Could we just call this "completion" ? Then it will match the request parameter, and we can use the same field for storing aggregate completions in the future once we implement that.

Reply:
👍